src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java

```diff
@@ -963,6 +969,10 @@ protected void setAttributes(Section mainSection) throws MojoExecutionException,
         if (compatibleSinceVersion!=null)
             mainSection.addAttributeAndCheck(new Attribute("Compatible-Since-Version", compatibleSinceVersion));
 
+        if (this.minimumJavaVersion != null) {
+            mainSection.addAttributeAndCheck(new Attribute("Minimum-Java-Version", this.minimumJavaVersion));
+        }
+
```

Comment:
Inclined to think we should actually probably issue a warning if not specified, also to advertise that feature.
I strongly fear that we'll have plugins de facto requiring Java 11 to run that will miss they can express this requirement without being nudged about it like here.

Reply:
@batmat I am ready to add validation if it helps.
Speaking of the borderline chekcs in general, https://github.com/jenkinsci/plugin-pom/pull/133 should add some enforcement on the Plugin POM side